### PR TITLE
feat: add chromestage

### PR DESCRIPTION
**Pull request by ajbouh**

### CODE REVIEW

File: images/chromestage/cmd/chromestage/proxy.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		return true // Adjust the origin checking to suit your needs
+	},
+}
+
+func novncHandler(w http.ResponseWriter, r *http.Request) {
+	ws, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		fmt.Println("Error upgrading to websocket:", err)
```
Comment (mgood) [inline images/chromestage/cmd/chromestage/proxy.go:22]: Maybe `log.Println` for these instead?

Comment (ajbouh) [reply images/chromestage/cmd/chromestage/proxy.go:22]: yes, agreed

File: images/chromestage/cmd/chromestage/commands.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+func (c *CDPCommander) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// GET returns commands (with meta header including url, update timestamp, revision id)
+	// POST runs a command (accepts meta header including url, update timestamp, revision id; errors on meta mismatch)
+	switch {
+	case r.Method == http.MethodGet:
+		h := w.Header()
+		h.Set("Content-Type", "application/json")
+		commands, err := c.commandIndex(c.ctx)
+		if err != nil {
+			http.Error(w, fmt.Sprintf(`{"error": %q}`, err.Error()), http.StatusInternalServerError)
```
[the remark this answers is not in the document]
Comment (ajbouh) [reply images/chromestage/cmd/chromestage/commands.go:147]: for now i think we want prioritize ease of debugging

**Comment by ajbouh** [comment]
I'm going to use rebase and merge to keep original chromestage commit from @progrium (since it's only additive and while incomplete it won't break anything else), and then layer the rest of my changes in a single subsequent commit.

Chromestage still needs a bit of work to meet all of our requirements, but it's good enough to enable the next round of experimentation. So we'll merge it now to enable forward progress.

Future work will:
- Add record/replay support
- Support sending a stream to bridge or other internal systems (eventually with audio)
- Support casting to devices like Chromecast or AppleTV
- Support sending audio to viewer

It's possible this will be built by continuing to use VNC or by switching to something like neko (which is based on WebRTC), selkies-gstreamer (based on gstreamer), or guacd (which would support RDP).

**Pull request merged**